### ISSUE TITLE
mpv does not play single frame video for correct duration

### ISSUE THREAD

**Issue by e00E** [mpv-player/mpv#8858]
### Important Information
Using the arch linux package:

```
> mpv --version
mpv 0.33.1-dirty Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
FFmpeg library versions:
   libavutil       56.70.100
   libavcodec      58.134.100
   libavformat     58.76.100
   libswscale      5.9.100
   libavfilter     7.110.100
   libswresample   3.9.100
FFmpeg version: n4.4
```

### Reproduction steps

```
mpv --no-config out.mp4
```

### Expected behavior

mpv plays the file for 10 seconds

### Actual behavior

mpv plays the file for roughly 1 second

### Log file

[log.txt](https://github.com/mpv-player/mpv/files/6528919/log.txt)

### Sample files

https://user-images.githubusercontent.com/1462179/119276673-b020a380-bc1b-11eb-8742-8ec4d64ac17b.mp4

The sample file has been created with ffmpeg:

```
ffmpeg -framerate 1/10 -i screenshot.png -c:v libx264 -pix_fmt yuv420p out.mp4
```

Both mpv and ffprobe show that the duration is 10 seconds. Firefox plays the file for the correct duration.

**Comment by ValZapod** [comment]
0.1 frames per second means that only 1 frame is present in 10 seconds and thus there is no point in showing it 10 seconds, after all EOF is already there.

> Firefox plays the file for the correct duration.

And so is chrome.

At fault is ffplay  -autoexit, I suppose.

**Comment by e00E** [comment]
> 0.1 frames per second means that only 1 frame is present in 10 seconds and thus there is no point in showing it 10 seconds, after all EOF is already there.

I disagree. The file's duration is 10 seconds. For those 10 seconds this frame is supposed to be shown. That is the point of the video file. There not being another future frame is no argument for not playing the file for the intended duration.
I am aware that this is quite an edge case that mpv might feel is not worth supporting but it is still a bug.

**Comment by ValZapod** [comment]
> The file's duration is 10 seconds.

That metadata is ofyen unreliable though.
